### PR TITLE
fix:[CDS-50511]: fix secret resolution for app name for PCF

### PR DESCRIPTION
**Pull request by sourabh-awashti**
## Describe your changes
Changes are related to secret app name not getting resolved in app setup step for pcf.
Tests Done
1. Running sanity pipelines for pcf
2. Running BG, basic and Canary deployment with app name as secret at account, org and project level.
3. Running rollback for all types of deployments with app name as secret.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/44529)

### CODE REVIEW

**Comment by sourabh-awashti** [comment]
trigger messagecheck

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 7 of 7 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44529)* status: 7 file(s) to review. 1 of 2 approvals obtained.

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 @OwnedBy(CDP)
 @EqualsAndHashCode(callSuper = true)
 public class CfDeployCommandRequestNG extends AbstractTasTaskRequest {
-  String newReleaseName;
+  @Expression(ALLOW_SECRETS) String newReleaseName;
   List<String> routeMaps;
```
Comment (RishabhGupta34) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java:39]: secret needed for routeMaps?

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRoutesRequestNG.java [hunk 2]
```diff
@@ -31,15 +33,15 @@
...
-  List<String> tempRoutes;
-  List<String> finalRoutes;
+  @Expression(ALLOW_SECRETS) String newApplicationName;
+  @Expression(ALLOW_SECRETS) List<String> tempRoutes;
+  @Expression(ALLOW_SECRETS) List<String> finalRoutes;
   boolean downsizeOldApplication;
   TasApplicationInfo activeApplicationDetails;
   TasApplicationInfo inActiveApplicationDetails;
   TasApplicationInfo newApplicationDetails;
   List<String> existingApplicationNames;
```
Comment (RishabhGupta34) [inline 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRoutesRequestNG.java:43]: secret needed for existingApplicationNames?

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRollbackCommandRequestNG.java [hunk 3]
```diff
@@ -33,11 +35,11 @@
...
-  List<String> routeMaps;
+  @Expression(ALLOW_SECRETS) List<String> routeMaps;
   TasApplicationInfo activeApplicationDetails;
   TasApplicationInfo newApplicationDetails;
   TasApplicationInfo inActiveApplicationDetails;
-  String cfAppNamePrefix;
+  @Expression(ALLOW_SECRETS) String cfAppNamePrefix;
   Integer activeAppRevision;
   boolean enforceSslValidation;
   boolean useAppAutoScalar;
```
[the remark this answers is not in the document]
Comment (RishabhGupta34) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRollbackCommandRequestNG.java:45]: there is a field tempRoutes in this DTO

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRoutesRequestNG.java (hunk 2 above)
Comment (sourabh-awashti) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRoutesRequestNG.java:43]: existing app names are being fetched from active app info from setup data which can not be a secret so no need to add it here

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java (hunk 1 above)
Comment (sourabh-awashti) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java:39]: this field in cfDeployCommandRequestNG is not being used anywhere , so not added it

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRollbackCommandRequestNG.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (sourabh-awashti) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfSwapRollbackCommandRequestNG.java:45]: added

File: 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java (hunk 1 above)
Comment (RishabhGupta34) [reply 950-delegate-tasks-beans/src/main/java/io/harness/delegate/task/pcf/request/CfDeployCommandRequestNG.java:39]: ok

**Review by RishabhGupta34 — APPROVED** [review]
> Reviewed 4 of 6 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44529)* status: 9 file(s) to review. 2. unresolved discussions. 1 of 2 approvals obtained.

**Review by anilchowdhury-harness — APPROVED** [review]
:lgtm:

> Reviewed 3 of 7 files at r1, 4 of 6 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44529#-:-NQ5DntO9slwR6IETKZc:bnfp4nl)* status: 2. unresolved discussions.

**Review by RishabhGupta34 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/44529)* status: :shipit: complete! ✓

**Pull request merged**